### PR TITLE
Allow redefining the default tick symbol from x to something else.

### PR DESCRIPTION
**Pull request by ronakg**
The title says it all.

This change also matches any 1 character between [] for undone.

### CODE REVIEW

File: plugin/simple-todo.vim [hunk 1]
```diff
@@ -46,14 +50,14 @@ nnore <Plug>(simple-todo-above) O<c-r>=<SID>get_list_marker(line('.')+1)<cr>[ ]<
...
+vnore <Plug>(simple-todo-mark-as-done) :execute 's/^\(\s*[-+*]\?\s*\)\[ \]/\1[' . g:simple_todo_tick_symbol . ']/'<cr>
+inore <Plug>(simple-todo-mark-as-done) <Esc>:execute 's/^\(\s*[-+*]\?\s*\)\[ \]/\1[' . g:simple_todo_tick_symbol . ']/'<cr>
 
 " Mark as undone
-nnore <Plug>(simple-todo-mark-as-undone) :s/^\(\s*[-+*]\?\s*\)\[x\]/\1[ ]/<cr>
-vnore <Plug>(simple-todo-mark-as-undone) :s/^\(\s*[-+*]\?\s*\)\[x\]/\1[ ]/<cr>
-inore <Plug>(simple-todo-mark-as-undone) <Esc>:s/^\(\s*[-+*]\?\s*\)\[x\]/\1[ ]/<cr>
+nnore <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)\[.\?\]/\1[ ]/'<cr>
+vnore <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)\[.\?\]/\1[ ]/'<cr>
+inore <Plug>(simple-todo-mark-as-undone) <Esc>:execute 's/^\(\s*[-+*]\?\s*\)\[.\?\]/\1[ ]/'<cr>
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Please explicitly specify the tick symbol in regexp, so the only items marked as done are highlighted and it's possible to navigate between them via `n`/`N` keys.

File: doc/simple-todo.txt [hunk 2]
```diff
@@ -11,7 +11,8 @@ Manage todo lists. Really simple.                                *simple-todo*
 
     1. Intro.................................|simple-todo-into|
     2. Mappings..............................|simple-todo-maps|
-    3. Contributing..........................|simple-todo-contrib|
+    3. Tick Symbol...........................|simple-todo-tick-symbol|
```
Comment (vitalk) [inline doc/simple-todo.txt]: Please use something more abstract, e.g. `Customization`.

File: plugin/simple-todo.vim (hunk 1 above)
Comment (ronakg) [inline plugin/simple-todo.vim]: So use the same variable ``g:simple_todo_tick_symbol' for mark as undone as well?

I didn't get the part about highlighting.

File: doc/simple-todo.txt (hunk 2 above)
Comment (ronakg) [inline doc/simple-todo.txt]: Will do.

File: plugin/simple-todo.vim (hunk 1 above)
Comment (vitalk) [inline plugin/simple-todo.vim]: Yes, use the similar pattern as for `<plug>(simple-todo-mark-as-done)`.

Comment (ronakg) [inline plugin/simple-todo.vim]: The reason I didn't do that is to allow different characters to be used as tick symbol.

Say someone is using x in their todo notes. Then after this update, they decide to use y as tick symbol. When they change the symbol to y, they won't be able to mark anything undone as the regex will fail.

Comment (vitalk) [inline plugin/simple-todo.vim]: Agreed, user should manually converts the tick symbols to new value, but I want to keep this behaviour:

- Item marked as done or undone
- This category (done or undone) highlighed
- The default search movements (`n`/`N`) work as expected, so it's possible to
  move to the next/previous item in that category.

**Comment by ronakg** [comment]
Take a look again.

**Comment by vitalk** [comment]
Looks good. Before I merge, please squash your commits.

**Comment by ronakg** [comment]
Done.

**Comment by vitalk** [comment]
@ronakg please squash your commits.

**Comment by ronakg** [comment]
I think I'm not doing this right (never done this before).

Should I close this request and raise a new pull request?

**Comment by vitalk** [comment]
No, just do `git rebase -i HEAD^3` (`3` is number of commits to process). Then mark the commits to squash, for example

```
pick 5c20cb7 Inherit list marker when create new todo at start of the line
f 0108bc4 Allow redefining the tick symbol to something else
f 77b1bf5 better readme
f 23e826e address comments from vitalk
```

When you save and close the editor, then the latest 3 comits merged into the 1st one.

Then push changes into repository, e.g. `git push origin master --force`.

**Comment by ronakg** [comment]
I'm getting this error - 

``$ git rebase -i HEAD~3
error: could not apply 0108bc4... Allow redefining the tick symbol to something else

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

Could not apply 0108bc472279c2506155b36dba2f53938044e6c1... Allow redefining the tick symbol to something else
``

**Comment by vitalk** [comment]
Looks like you already squashed them into 4b0287c.

This error occurs because of your latest merge commit b1f0ed7, e.g.

```
*   b1f0ed7 - (HEAD, ronakg/master) Merge branch 'master' of github.com:ronakg/vim-simple-todo (9 hours ago) <Ronak Gandhi>
|\
| * ee2d508 - Capitalize plug (9 hours ago) <Ronak Gandhi>
| * 23e826e - address comments from vitalk (9 hours ago) <Ronak Gandhi>
| * 77b1bf5 - better readme (12 hours ago) <Ronak Gandhi>
| * 0108bc4 - Allow redefining the tick symbol to something else (12 hours ago) <Ronak Gandhi>
* | 4b0287c - Allow redefining the tick symbol to something else (9 hours ago) <Ronak Gandhi>
|/
* 5c20cb7 - (tag: 0.12, origin/master, master) Inherit list marker when create new todo at start of the line (5 weeks ago) <Vital Kudzelka>
*
```

Just remove it `git reset --hard HEAD^`

```
* 4b0287c - (HEAD) Allow redefining the tick symbol to something else (9 hours ago) <Ronak Gandhi>
* 5c20cb7 - (tag: 0.12, origin/master, master) Inherit list marker when create new todo at start of the line (5 weeks ago) <Vital Kudzelka>
```

Now rewrite commit message `git commit --amend`, because 4b0287c contains commit messages from your previous commits, e.g.

```
Allow redefining the tick symbol to something else

better readme

address comments from vitalk

Capitalize plug
```

After that you have a more clean history with single, well-written commit.
Push changes back to repository, e.g. `git push origin --force`.

**Comment by ronakg** [comment]
Done. Thanks for detailed steps.

**Comment by vitalk** [comment]
Well done! Thank you for your contribution.

**Pull request merged**